### PR TITLE
[Improve] [spark-source-kudu] Remove repeated format function call in spark-source-kudu

### LINKED ISSUE #1458: [Improve] [spark-source-kudu] Remove repeated format function in Kudu source
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

I found `org.apache.seatunnel.spark.source.Kudu` have a repeated `org.apache.spark.sql.DataFrameReader#format` function call due to an implicit function when read data from kudu using spark, just like below:  
```scala
    val ds = env.getSparkSession.read
      .format("org.apache.kudu.spark.kudu")
      .options(mapConf) 
      .kudu // Notice, here is an implicit function
```
The implicit function like below:  
```scala
  implicit class KuduDataFrameReader(reader: DataFrameReader) {
    def kudu: DataFrame = reader.format("org.apache.kudu.spark.kudu").load
  }
```

So the `format("org.apache.kudu.spark.kudu")` is called twice.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Rianico**
## Purpose of this pull request
closed #1458 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala [hunk 1]
```diff
@@ -36,7 +36,6 @@ class Kudu extends SparkBatchSource {
       "kudu.table" -> config.getString("kudu_table"))
 
     val ds = env.getSparkSession.read
-      .format("org.apache.kudu.spark.kudu")
       .options(mapConf)
       .kudu
```
Comment (wuchunfu) [inline seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: @Rianico Seems like this way is better
```scala
// Create a DataFrame that points to the Kudu table we want to query.
val df = spark.read.options(Map("kudu.master" -> "kudu.master:7051",
                                 "kudu.table" -> "default.my_table")).format("kudu").load
```

Please refer to: https://kudu.apache.org/docs/developing.html

Comment (Rianico) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: `format("kudu")` is implemented by `org.apache.kudu:kudu-spark2_2.11`. But it only take affect from `1.9.0`, and earlier version will get an `java.lang.ClassNotFoundException: kudu.DefaultSource` Exception.   
And the version that I see in pom is `1.7.0` :
```xml
<kudu-spark.version>1.7.0</kudu-spark.version>
```
So it is better to use `kudu` function directly, we don't need to pay attention about `format("kudu")` or `format("org.apache.kudu.spark.kudu")`.
  
Consider the situation if we have many spark options, the Kudu's official code example may look not so clear.

**Review by simon824 — APPROVED** [review]
LGTM

**Review by Rianico — COMMENTED** [review]
Remove repeated format function call in spark-source-kudu

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala (hunk 1 above)
Comment (yx91490) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: how about upgrade the kudu-spark version? 1.7.0 seems a little old.

Comment (wuchunfu) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: > how about upgrade the kudu-spark version? 1.7.0 seems a little old.

+1

Comment (Rianico) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: > how about upgrade the kudu-spark version? 1.7.0 seems a little old.  

+1. How about `1.12.0` ? I have saw that many problems which belong to high priority in jira was solved in this version. 
There is also two key point about Kudu's version selection:  
- From `1.9.0`, Kudu offers a way to do junit test, we can refer to: [testing-apache-kudu-applications-on-the-jvm](https://kudu.apache.org/2019/03/19/testing-apache-kudu-applications-on-the-jvm.html#testing-apache-kudu-applications-on-the-jvm).
- From `1.10.0`, Kudu supports both full and incremental table backups via a job implemented using Apache Spark. We can expand more feature about kudu if we use the version from than `1.10.0`. But maybe we should select the version newer than `1.10.1` due to [KUDU-2990](https://issues.apache.org/jira/browse/KUDU-2990).

Comment (yx91490) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: vote for 1.12.0

Comment (Rianico) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: @wuchunfu How about this idea? I'm also willing to contribute the code about the upgrade of Kudu's version and the addition of unit test. :smile: But this is another topic and maybe  I should create another issue and PR?

**Review by CalvinKirs — APPROVED** [review]

File: seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala (hunk 1 above)
Comment (CalvinKirs) [reply seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-kudu/src/main/scala/org/apache/seatunnel/spark/source/Kudu.scala:40]: It's better to create an email thread to discuss,

**Pull request merged**